packages/styled-components/src/models/StyledComponent.js

```diff
@@ -147,7 +147,7 @@ class StyledComponent extends Component<*> {
 
       if (key === 'forwardedComponent' || key === 'as') continue;
       else if (key === 'forwardedRef') propsForElement.ref = computedProps[key];
-      else if (!isTargetTag || validAttr(key)) {
+      else if ((!isTargetTag && key !== 'suppressClassNameWarning') || validAttr(key)) {
```

Comment:
can you move this up to L148 where `forwardedComponent` and `as` are? It can just be skipped along with those known props

Reply:
good point, done!